ntfysh_client/NotificationDialog.cs

```diff
@@ -0,0 +1,263 @@
+        private static extern bool AnimateWindow(IntPtr hWnd, int time, int flags);
+
+        private const int ScreenMargin = 20;
+
+        private int _timeout = 0;
+        private System.Timers.Timer? displayTimeoutTimer = null;
+        private System.Windows.Forms.Timer? updateTimer = null;
+        private Stopwatch? shownStopwatch = null;
+        private ToolTipIcon? _icon;
+        private int _progress_value = 0;
+        private int progress
+        {
+            get { return this._progress_value; }
+            set
+            {
+                this._progress_value = value;
+                this.progressBar1.Value = value;
+            }
+        }
+
+        public bool IsVisible
+        {
+            get { return this.Visible; }
+            set { this.Visible = value; }
+        }
+
+        public NotificationDialog()
+        {
+            this.ShowInTaskbar = false;
+            this.IsVisible = false;
+            this.TopMost = true;
+            InitializeComponent();
+            InitializeWindowHidden();
+        }
+
+        public void ShowNotification(string title, string message, int timeout_ms = -1, ToolTipIcon? icon = null, bool showTimeOutBar = true, bool showInDarkMode = true)
+        {
+            if (this.IsVisible)
+            {
+                // close the current notification
+                this.handleTimeout(null, null);
+            }
+            // setup data
+            this._icon = icon;
+            if (this._icon != null)
+            {
+                this.iconBox.Image = ConvertToolTipIconToImage(_icon.Value);
+            }
+            this.tbTitle.Text = title;
+            this.tbMessage.Text = message;
+
+            // setup timers
+            if (this.displayTimeoutTimer != null)
+            {
+                this.displayTimeoutTimer.Stop();
+                this.displayTimeoutTimer.Dispose();
+            }
+            if (this.updateTimer != null)
+            {
+                this.updateTimer.Stop();
+                this.updateTimer.Dispose();
+            }
+            if (timeout_ms > 0)
+            {
+                this.displayTimeoutTimer = new System.Timers.Timer(timeout_ms);
+                displayTimeoutTimer.Elapsed += handleTimeout;
+                this.displayTimeoutTimer.Start();
+
+                if (showTimeOutBar)
+                {
+                    this.progress = 100;
+                    this.updateTimer = new System.Windows.Forms.Timer();
+                    updateTimer.Interval = 100;
+                    this.updateTimer.Tick += this.UpdateProgress;
+                    this.updateTimer.Start();
+
+                    this.shownStopwatch = new Stopwatch();
+                    this.shownStopwatch.Start();
+
+                    this.progressBar1.Visible = true;
+                    this.lbTimeout.Visible = true;
+                    this._timeout = timeout_ms;
+                } else
```

Comment:
Nitpick: newline

Reply:
heh, wonder where that style came from...